tests/flytekit/unit/sdk/tasks/test_dynamic_tasks.py

```diff
@@ -109,6 +110,39 @@ def no_future_batch_task(wf_params, in1, out_str):
     out_str.set(["res1", "res2"])
 
 
+def manual_assign_name():
+    pass
+
+
+@inputs(task_input_num=Types.Integer)
+@outputs(out=Types.Integer)
+@dynamic_task
+def dynamic_wf_task(wf_params, task_input_num, out):
+    wf_params.logging.info("Running inner task... yielding a code generated sub workflow")
+
+    input_a = Input(Types.Integer, help="Tell me something")
+    node1 = sq_sub_task(in1=input_a)
+
+    MyUnregisteredWorkflow = workflow(
+        inputs={
+            'a': input_a,
+        },
+        outputs={
+            'ooo': Output(node1.outputs.out1, sdk_type=Types.Integer,
```

Comment:
do things still work if this line isn't here?  that is, if the sub workflow runs a task for side-effect, and has no outputs, and hence doesn't pass any outputs along through the workflow, will that task still be seen as an upstream entity?

Reply:
Added a unit test below to test that... seem to work..